src/mito2/src/read/scan_util.rs

```diff
@@ -326,6 +409,53 @@ impl fmt::Debug for ScanMetricsSet {
             write!(f, ", \"dedup_metrics\":{:?}", dedup_metrics)?;
         }
 
+        // Write top file metrics if present and non-empty
+        if let Some(file_metrics) = per_file_metrics
+            && !file_metrics.is_empty()
+        {
+            // Use min-heap (BinaryHeap with reverse comparison) to keep only top 10
+            let mut heap = BinaryHeap::new();
+            for (file_id, metrics) in file_metrics.iter() {
+                let total_cost =
+                    metrics.build_part_cost + metrics.build_reader_cost + metrics.scan_cost;
+
+                if heap.len() < 10 {
+                    // Haven't reached 10 yet, just push
+                    heap.push(CompareCostReverse {
+                        total_cost,
+                        file_id: *file_id,
+                        metrics,
+                    });
+                } else if let Some(min_entry) = heap.peek() {
+                    // If current cost is higher than the minimum in our top-10, replace it
+                    if total_cost > min_entry.total_cost {
+                        heap.pop();
+                        heap.push(CompareCostReverse {
+                            total_cost,
+                            file_id: *file_id,
+                            metrics,
+                        });
+                    }
+                }
+            }
+
+            // Pop all items and collect (they come out in ascending order)
+            let top_files: Vec<_> = heap
+                .into_iter()
+                .map(|entry| (entry.file_id, entry.metrics))
+                .collect();
```

Comment:
to get a ordered output you may want to use https://doc.rust-lang.org/std/collections/struct.BinaryHeap.html#method.into_iter_sorted or `into_sorted_vec()`